modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleServiceIT.java

```diff
@@ -263,12 +231,12 @@ public void testSystemDataStreamRetention() throws Exception {
                     ).actionGet();
                     assertThat(getDataStreamResponse.getDataStreams().size(), equalTo(1));
                     assertThat(getDataStreamResponse.getDataStreams().get(0).getDataStream().getName(), equalTo(SYSTEM_DATA_STREAM_NAME));
-                    List<Index> backingIndices = getDataStreamResponse.getDataStreams().get(0).getDataStream().getIndices();
-                    assertThat(backingIndices.size(), equalTo(1)); // global retention is ignored
+                    List<Index> currentBackingIndices = getDataStreamResponse.getDataStreams().get(0).getDataStream().getIndices();
```

Comment:
We could use `getDataStreamBackingIndexNames(SYSTEM_DATA_STREAM_NAME)` here, right?

Reply:
We're serializing the `getDataStreamResponse` to XContent a few lines below, so I preferred this over doing two API calls.